hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/ExpressionEvalContext.java

```diff
@@ -67,7 +77,25 @@ static ExpressionEvalContext from(Context ctx) {
             return new ExpressionEvalContextImpl(
                     arguments,
                     new DefaultSerializationServiceBuilder().build(),
-                    Util.getNodeEngine(ctx.hazelcastInstance()));
+                    Util.getNodeEngine(ctx.hazelcastInstance()),
+                    NoOpSqlSecurityContext.INSTANCE);
+        }
+    }
+
+    static ExpressionEvalContext recreateContext(
+            @Nullable ExpressionEvalContext eec,
+            @Nonnull HazelcastInstance hz,
+            @Nonnull List<Object> arguments,
+            @Nonnull InternalSerializationService iss,
+            @Nullable Subject subject) {
+        if (eec == null) {
+            NodeEngineImpl nodeEngine = Util.getNodeEngine(hz);
+            SecurityContext sc = nodeEngine.getNode().securityContext;
+            SqlSecurityContext ssc = sc != null ? sc.createSqlContext(subject) : NoOpSqlSecurityContext.INSTANCE;
+            return new ExpressionEvalContextImpl(arguments, iss, nodeEngine, ssc);
+        } else {
+            ExpressionEvalContextImpl eeci = (ExpressionEvalContextImpl) eec;
```

Comment:
in what scenarios this branch is taken and how can HZ instance or serialization service change for EECI?

Reply:
E.g, `MapOperation`

Reply:
if we use serialization (operation sent to different member), then EEC is created from scratch, if we don't (operation on the same member) then nodeEngine and serialization service will be the same (tricky part: job-specific serialization serivce). 
This was not needed before, why is it needed now?

Reply:
> This was not needed before

Emmm, no? It was needed before, at least, for serialization purposes: when the entry processor is initializing on this member (not coordinator), it should know about the data format it will handle.

Reply:
> This was not needed before

"this" = what you called "recreateContext". We always created EEC, never updated node engine or serialization sevice in existing one.

Reply:
Hmm, I thought about it. I will revert the bevaviour.

Reply:
https://github.com/hazelcast/hazelcast/pull/25477/commits/531ce9d168aee5b0502fe69d8adb2c42bbad76ab